src/function/fetch.rs

```diff
@@ -18,13 +18,13 @@ where
         zalsa.unwind_if_revision_cancelled(db);
 
         let memo = self.refresh_memo(db, id);
-        // SAFETY: We just refreshed the memo so it is guaranteed to contain a value now.
         let StampedValue {
             value,
             durability,
             changed_at,
         } = memo
             .revisions
+            // SAFETY: We just refreshed the memo so it is guaranteed to contain a value now.
             .stamped_value(unsafe { memo.value.as_ref().unwrap_unchecked() });
```

Comment:
```suggestion
        let memo = self.refresh_memo(db, id);
        // SAFETY: We just refreshed the memo so it is guaranteed to contain a value now.
        let stamped_value = memo
            .revisions
            .stamped_value(unsafe { memo.value.as_ref().unwrap_unchecked() });
        let StampedValue {
            value,
            durability,
            changed_at,
        } = stamped_value;
```
That way we keep the comment close to the relevant part without interrupting the flow

Reply:
That would still trigger the lint, but pulling out the argument instead does the trick.

Reply:
Oh does clippy specifically check by following line? I see. Well, your change is better either way!